### PR TITLE
v5.0.1

### PR DESCRIPTION
**Pull request by flaviendelangle**
We need a new release to apply https://github.com/mui-org/material-ui-x/pull/3196
But as @m4theushw told me, we probably need to do a full release with everything in between otherwise the PR we merged since will be on the tag but not on the NPM release.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,20 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## 5.0.1
+
+_Nov 16, 2021_
+
+A big thanks to the 1 contributor who made this release possible.
+
+### `@mui/x-data-grid@v5.0.0-beta.1` / `@mui/x-data-grid-pro@v5.0.0-beta.1`
```
Comment (m4theushw) [inline CHANGELOG.md:12]:
```suggestion
### `@mui/x-data-grid@v5.0.1` / `@mui/x-data-grid-pro@v5.0.1`
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,64 @@
...
+      preProcessEditCellProps: async (params: GridEditCellPropsChangeParams) => {
+        const userWithEmail = await fetchUserByEmail(params.value);
+        const hasError = !!userWithEmail;
+        return { ...params.props, error: hasError };
+      }
+    }
+  ];
+  ```
+
+- ✨ New method `getRootDimensions` to access the size of the grid #3007 @flaviendelangle
```
Comment (m4theushw) [inline CHANGELOG.md:36]:
```suggestion
- ✨ New method `getRootDimensions` to access the size of the grid (#3007) @flaviendelangle
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,64 @@
...
+
+  It contains the size of the viewport (which is the scrollable `div` containing the rows) both with and without the scrollbars.
```
Comment (m4theushw) [inline CHANGELOG.md:38]:
```suggestion
  It contains the size of the viewport (which is the scrollable container containing the rows and columns) considering scrollbars or not.
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,64 @@
...
+  ```ts
+  const dimensions = apiRef.current.getRootDimensions();
+  ```
+
+### `@mui/x-data-grid@v5.0.1` / `@mui/x-data-grid-pro@v5.0.1`
+
+#### Changes
+
+- [DataGrid] New API to validate the editing values (#3006) @m4theushw
+- [DataGrid] Use color-scheme to set dark mode on native components (#3146 @alexfauquette
```
Comment (m4theushw) [inline CHANGELOG.md:49]:
```suggestion
- [DataGrid] Use color-scheme to set dark mode on native components (#3146) @alexfauquette
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,6 +3,64 @@
...
+
+- [core] Drop `useGridContainerProps` (#3007) @flaviendelangle
+- [core] Move `getRowIdFromRowIndex` and `getRowIndex` to the sorting API (#3126) @flaviendelangle
+- [core] Polish v5 CHANGELOG (#3194) @oliviertassinari
+- [core] Remove the `index.ts` of the export hooks (#3165) @flaviendelangle
+- [core] Set the correct release date for v5.0.0 in the CHANGELOG.md (#3192) @flaviendelangle
+
+### Docs
+
+- [docs] move sentence about disabling multi rows selection (#3167) @alexfauquette
```
Comment (m4theushw) [inline CHANGELOG.md:62]:
```suggestion
- [docs] Move sentence about disabling multi rows selection (#3167) @alexfauquette
```

**Review by m4theushw — APPROVED** [review]

File: CHANGELOG.md [hunk 6]
```diff
@@ -3,6 +3,64 @@
...
+
+### `@mui/x-data-grid@v5.0.1` / `@mui/x-data-grid-pro@v5.0.1`
+
+#### Changes
+
+- [DataGrid] New API to validate the editing values (#3006) @m4theushw
+- [DataGrid] Use color-scheme to set dark mode on native components (#3146) @alexfauquette
+- [DataGrid] Fix the `@mui/x-data-grid` type entrypoint (#3196) @flaviendelangle
+
+### Core
```
Comment (oliviertassinari) [inline CHANGELOG.md:52]: I think that "Docs" should come before "Core". I think that changes in "Core" are most interesting for maintainers.

**Pull request merged**